reuseport_plan9.go

```diff
@@ -0,0 +1,39 @@
+package tcpreuse
+
+import (
+	"net"
+	"os"
+)
+
+// reuseErrShouldRetry diagnoses whether to retry after a reuse error.
+// if we failed to bind, we should retry. if bind worked and this is a
+// real dial error (remote end didnt answer) then we should not retry.
+func reuseErrShouldRetry(err error) bool {
+	if err == nil {
+		return false // hey, it worked! no need to retry.
+	}
+
+	// if it's a network timeout error, it's a legitimate failure.
+	if nerr, ok := err.(net.Error); ok && nerr.Timeout() {
+		return false
+	}
+
+	e, ok := err.(*net.OpError)
+	if !ok {
+		return true
+	}
+
+	e1, ok := e.Err.(*os.PathError)
+	if !ok {
+		return true
+	}
+
+	switch e1.Err.Error() {
+	case "address in use":
```

Comment:
define as constants? or are these defined in the golang plan9 syscall package?

Reply:
There are [few such errors](https://github.com/golang/go/blob/master/src/syscall/errors_plan9.go#L27) in the syscall package, but not this one. I could try to convince upstream to add these errors, but it's not going to help us because `reuseport_posix.go` does a type assertion on `syscall.Errno`, which doesn't exist on Plan 9 (it uses `syscall.ErrorString`). I'll define the constants in this file.

Reply:
in that file seems fine

Reply:
Done